diff --git a/src/devices/system.c b/src/devices/system.c
index edd281e..1cb5000 100644
--- a/src/devices/system.c
+++ b/src/devices/system.c
@@ -36,7 +36,7 @@ static void
 system_print(Stack *s)
 {
 	Uint8 i;
-	for(i = s->ptr - 7; i != (Uint8)(s->ptr + 1); i++)
+	for(i = s->ptr - 15; i != (Uint8)(s->ptr + 1); i++)
 		fprintf(stderr, "%02x%c", s->dat[i], i == 0 ? '|' : ' ');
 	fprintf(stderr, "< \n");
 }
